src/niri.rs

```diff
@@ -898,10 +914,9 @@ impl State {
 
         let pointer = &self.niri.seat.get_pointer().unwrap();
         let location = pointer.current_location();
-        let under = if self.niri.pointer_hidden {
-            PointContents::default()
-        } else {
-            self.niri.contents_under(location)
+        let under = match self.niri.pointer_visibility {
+            PointerVisibility::Disabled => PointContents::default(),
+            _ => self.niri.contents_under(location),
```

Comment:
Hey, so this looks good, but I also wanted the pointer to go to Disabled once contents_under() changes. I.e. if you scroll the view or switch the workspace via keyboard (so pointer now has something different underneath), then it should go to Disabled (because it's invisible).